satellite/console/consoleweb/server.go

```diff
@@ -192,6 +217,323 @@ func (server *Server) appHandler(w http.ResponseWriter, r *http.Request) {
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		w.WriteHeader(400)
+		return
+	}
+
+	err = server.service.DeleteAccount(ctx, password.Password)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		return
+	}
+
+	err = json.NewEncoder(w).Encode(auth.User)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		w.WriteHeader(500)
+		server.log.Debug("Error serializing response: " + err.Error())
+	}
+}
+
+// resendEmailRequestHandler resend activation email for given email
+func (server *Server) resendEmailRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+	params := mux.Vars(r)
+	val, ok := params["id"]
+	if !ok {
+		err = errs.New("id expected")
+
+		server.serveJsonError(w, 400, err)
+		return
+	}
+
+	userID, err := uuid.Parse(val)
+	if err != nil {
+		server.serveJsonError(w, 400, err)
+		return
+	}
+
+	user, err := server.service.GetUser(ctx, *userID)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		return
+	}
+	token, err := server.service.GenerateActivationToken(ctx, user.ID, user.Email)
+
+	rootObject, ok := ctx.Value("rootObject").(RootObject)
+	if !ok {
+		server.log.Error("root object is not set")
+		return
+	}
+	link := rootObject.Origin + rootObject.ActivationPath + token
+	userName := user.ShortName
+	if user.ShortName == "" {
+		userName = user.FullName
+	}
+
+	contactInfoURL := rootObject.ContactInfoURL
+	termsAndConditionsURL := rootObject.TermsAndConditionsURL
+
+	server.mailService.SendRenderedAsync(
+		ctx,
+		[]post.Address{{Address: user.Email, Name: userName}},
+		&consoleql.AccountActivationEmail{
+			Origin:                rootObject.Origin,
+			ActivationLink:        link,
+			TermsAndConditionsURL: termsAndConditionsURL,
+			ContactInfoURL:        contactInfoURL,
+		},
+	)
+}
+
+// forgotPasswordRequestHandler creates reset password token and send user email
+func (server *Server) forgotPasswordRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	params := mux.Vars(r)
+	email, ok := params["email"]
+	if !ok {
+		err = errs.New("email expected")
+
+		server.serveJsonError(w, 400, err)
+		return
+	}
+
+	user, err := server.service.GetUserByEmail(ctx, email)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		return
+	}
+
+	recoveryToken, err := server.service.GeneratePasswordRecoveryToken(ctx, user.ID)
+	if err != nil {
+		server.serveJsonError(w, 500, errs.New("failed to generate password recovery token"))
+	}
+
+	rootObject := ctx.Value("rootObject").(RootObject)
+	passwordRecoveryLink := rootObject.Origin + rootObject.PasswordRecoveryPath + recoveryToken
+	cancelPasswordRecoveryLink := rootObject.Origin + rootObject.CancelPasswordRecoveryPath + recoveryToken
+	userName := user.ShortName
+	if user.ShortName == "" {
+		userName = user.FullName
+	}
+
+	contactInfoURL := rootObject.ContactInfoURL
```

Comment:
Same you have access to server props here

Reply:
Removed